yt/visualization/plot_window.py

```diff
@@ -2268,17 +2268,23 @@ def SlicePlot(ds, normal=None, fields=None, axis=None, *args, **kwargs):
     ...                 north_vector=[0.2,-0.3,0.1])
 
     """
-    # Make sure we are passed a normal
-    # we check the axis keyword for backwards compatibility
-    if normal is None:
+    if axis is not None:
+        issue_deprecation_warning(
+            "SlicePlot's argument 'axis' is a deprecated alias for 'normal', it "
+            "will be removed in a future version of yt."
+        )
```

Comment:
Hmmm.  I have mixed feelings about this.

Reply:
What is wrong with it ? I feel like I'm missing something entirely.

Reply:
Well, I'll say that it's not your fault, I just have mixed feelings with encouraging folks to supply a normal vector.  I know that internally it's the same, but I think we made it that way so that you could swap out `"x"` and `[0.2, 0.3, 0.4]` or something without changing the argument name.

Reply:
So you mean this argument should be positional ? In python 3.8 positional-only arguments are a thing, but in the mean time, a different approach could be to remove the default value for normal, making it mandatory and encouraging the positional-arg syntax. The trade-off here is that it can not be done while also supporting backward compatibility.
While I'm all about removing dead code where I can, it was never signaled to users that the old behaviour was discouraged/deprecated, hence my choice of going for a smooth deprecation path. What do you think ?

Reply:
If I read the code correctly, it still is possible to call `yt.SlicePlot(ds, whatever, 'density')` with `whatever='x'` or `whatever=[0.2,0.3,0.4]`. @neutrinoceros am I wrong?

Reply:
Absolutely ! (no, you're not wrong)
I'm only trying to handle the case where the keyword `axis` is used instead of `normal`, but for scripts that write them positionally, it shouldn't change anything !

Reply:
So I'm ok with the way it is coded right now. @matthewturk is it good to go for you? I'm ok to merge

Reply:
lgtm